qrand/platforms/qsharp/job.py

```diff
@@ -31,34 +34,103 @@
 class QsharpJob(QuantumJob):
-    def __init__(self) -> None:
-        self.ERROR_MSG = f"{self.__class__.__name__}"  # TODO
-        raise NotImplementedError(self.ERROR_MSG)
+    def __init__(
+        self,
+        circuit: QsharpCircuit,
+        backend: QsharpBackend,
+        num_measurements: Optional[int] = None,
+    ) -> None:
+        self.backend: QsharpBackend = backend
+        self.circuit: QsharpCircuit = circuit
+        self.num_measurements: int = num_measurements  # type: ignore
 
     ############################### PUBLIC API ###############################
     @property
     def backend(self) -> QsharpBackend:
-        raise NotImplementedError(self.ERROR_MSG)
+        return self._backend
 
     @backend.setter
     def backend(self, backend: QsharpBackend) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        self._backend: QsharpBackend = backend
 
     @property
     def circuit(self) -> QsharpCircuit:
-        raise NotImplementedError(self.ERROR_MSG)
+        return self._circuit
 
     @circuit.setter
     def circuit(self, circuit: QsharpCircuit) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        if self.backend.max_qubits < circuit.num_qubits:
+            raise RuntimeError(
+                f"Failed to assign QsharpCircuit for QsharpJob. Number of \
+                qubits in QsharpCircuit unsupported by this job's Backend: \
+                {self.backend.max_qubits}<{circuit.num_qubits}."
+            )
+        self._circuit: QsharpCircuit = circuit
 
     @property
-    def num_measurements(self) -> int:
-        raise NotImplementedError(self.ERROR_MSG)
+    def num_measurements(self) -> int:  # type: ignore
+        return self._num_measurements
 
     @num_measurements.setter
     def num_measurements(self, num_measurements: Optional[int]) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        num_measurements = (
+            num_measurements
+            if isinstance(num_measurements, int) and 0 < num_measurements
+            else self.backend.max_measurements
+        )
+
+        if self.backend.max_measurements < num_measurements:
+            warn(
+                f"Number of measurements unsupported by the job's Backend: \
+                {self.backend.max_measurements}<{num_measurements}. \
+                Using max_measurements instead.",
+                UserWarning,
+            )
+            num_measurements = self.backend.max_measurements
+        self._num_measurements = num_measurements
+
+    def _generate_code(self) -> QSharpCallable:
+        qsharp_code = """
+        open Microsoft.Quantum.Intrinsic;
+        open Microsoft.Quantum.Measurement;
+        open Microsoft.Quantum.Arrays;
+        open Microsoft.Quantum.Math;
+
+        operation Program():String[]{{
+
+        use q = Qubit[{num_qubits}];
+        mutable value = ConstantArray({shots},"");
+        mutable res = "";
+
+        for i in IndexRange(value)
```

Comment:
So, if you are including the `shots` here by means of a for loop, then the `shots` argument when you call `azure.execute` should be modified accordingly (e.g. set to 1?).

Reply:
Set `shots` in `azure.execute` to 1